### PR TITLE
Feature/delete target

### PR DESCRIPTION
**Pull request by LeandroSoria2**
### What changes does this PR include?
**Description**
Add delete controller and RSpec test
**Summary**
- Add controller 
- Add delete test 

[Trello](https://trello.com/c/WtNGQzxF/6-2-as-a-user-i-should-be-able-to-delete-a-previously-created-target-so-its-no-longer-available-in-the-map)

### CODE REVIEW

**Review by JulianPasquale — APPROVED** [review]

File: app/controllers/api/v1/targets_controller.rb [hunk 1]
```diff
@@ -13,11 +13,24 @@ def create
         }, status: :unprocessable_entity
       end
 
+      def destroy
+        target.destroy!
+        head :no_content
+      rescue ActiveRecord::RecordNotFound => e
```
Comment (AgostinaDev) [inline app/controllers/api/v1/targets_controller.rb]: Try to rescue from exceptions in the ApiController & avoid handling them here

File: .reek.yml [hunk 2]
```diff
@@ -128,3 +128,4 @@ exclude_paths:
   - config
   - db/migrate
   - vendor
+  - app/controllers/api/v1/targets_controller.rb
```
Comment (AgostinaDev) [inline .reek.yml]: I guess this will be removed once adjusted in the 'create' branch

File: spec/requests/api/v1/targets/destroy_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    it 'deletes the target' do
+      expect {
+        subject
+      }.to change { Target.count }.by(-1)
+    end
+  end
+
+  context 'when user sends someone elses target id' do
+    let(:target_id) { 1237 }
```
Comment (AgostinaDev) [inline spec/requests/api/v1/targets/destroy_spec.rb]: How are you so sure that 1237 is someone else's target_id?

**Comment by juanmferreira93** [comment]
change the base to development and then rebase again :)

File: db/schema.rb [hunk 4]
```diff
@@ -63,7 +63,10 @@
     t.decimal "latitude", null: false
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+<<<<<<< HEAD
```
Comment (AgostinaDev) [inline db/schema.rb:66]: Looks like you had conflicts here

File: config/routes.rb [hunk 5]
```diff
@@ -12,7 +12,11 @@
     namespace :v1, defaults: { format: :json } do
       devise_scope :user do
         resources :topics, only: :index
+<<<<<<< HEAD
```
Comment (AgostinaDev) [inline config/routes.rb:15]: Looks like you had conflicts here

File: app/models/target.rb [hunk 6]
```diff
@@ -14,7 +14,10 @@
 #
 # Indexes
 #
+<<<<<<< HEAD
```
Comment (AgostinaDev) [inline app/models/target.rb]: Looks like you had conflicts here

File: config/routes.rb (hunk 5 above)
Comment (AgostinaDev) [inline config/routes.rb]: Looks like you had conflicts here

File: db/schema.rb (hunk 4 above)
Comment (AgostinaDev) [inline db/schema.rb]: Looks like you had conflicts here

File: spec/requests/api/v1/targets/create_spec.rb [hunk 7]
```diff
@@ -86,7 +86,6 @@
 
     context 'when user is not logged in' do
       let(:headers) { {} }
-
```
Comment (AgostinaDev) [inline spec/requests/api/v1/targets/create_spec.rb]: Please don't remove this empty line

File: app/views/api/v1/topics/targets/index.json.jbuilder [hunk 8]
```diff
@@ -0,0 +1 @@
+json.array! @targets, partial: 'api/v1/targets/info', as: :target
```
Comment (juanmferreira93) [inline app/views/api/v1/topics/targets/index.json.jbuilder]: why you have this view in this PR?

File: app/views/api/v1/topics/targets/create.json.jbuilder [hunk 9]
```diff
@@ -0,0 +1,3 @@
+json.target do
+  json.partial! 'api/v1/targets/info', target: @target
+end
```
Comment (juanmferreira93) [inline app/views/api/v1/topics/targets/create.json.jbuilder]: why you have this view in this PR?

File: app/views/api/v1/topics/targets/destroy.json.jbuilder [hunk 10]
```diff
@@ -0,0 +1 @@
+json.partial! 'api/v1/targets/info', target: target
```
Comment (juanmferreira93) [inline app/views/api/v1/topics/targets/destroy.json.jbuilder]: we do not need a view for destroy.. we just return `head :no_content` and that's enough

File: spec/requests/api/v1/targets/destroy_spec.rb (hunk 3 above)
Comment (juanmferreira93) [reply spec/requests/api/v1/targets/destroy_spec.rb]: you can use something random here

**Review by juanmferreira93 — COMMENTED** [review]
Please check that you already properly rebased this branch with development, and also has to be careful, some of the comments I left, we already discussed offline.

File: .reek.yml (hunk 2 above)
Comment (LeandroSoria2) [reply .reek.yml]: This was fixed in the last commit

File: spec/requests/api/v1/targets/destroy_spec.rb (hunk 3 above)
Comment (LeandroSoria2) [reply spec/requests/api/v1/targets/destroy_spec.rb]: This was fixed in last commit

File: app/models/target.rb (hunk 6 above)
Comment (LeandroSoria2) [reply app/models/target.rb]: It was fixed

File: config/routes.rb (hunk 5 above)
Comment (LeandroSoria2) [reply config/routes.rb]: It was fixed

File: db/schema.rb (hunk 4 above)
Comment (LeandroSoria2) [reply db/schema.rb]: It was fixed

File: spec/requests/api/v1/targets/create_spec.rb (hunk 7 above)
Comment (LeandroSoria2) [reply spec/requests/api/v1/targets/create_spec.rb]: It was fixed

File: app/views/api/v1/topics/targets/index.json.jbuilder (hunk 8 above)
Comment (LeandroSoria2) [reply app/views/api/v1/topics/targets/index.json.jbuilder]: It was fixed

File: app/views/api/v1/topics/targets/create.json.jbuilder (hunk 9 above)
Comment (LeandroSoria2) [reply app/views/api/v1/topics/targets/create.json.jbuilder]: this was fixed in the last commit

File: app/views/api/v1/topics/targets/destroy.json.jbuilder (hunk 10 above)
Comment (LeandroSoria2) [reply app/views/api/v1/topics/targets/destroy.json.jbuilder]: this was fixed in the last commit

File: spec/requests/api/v1/targets/destroy_spec.rb [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    it 'returns ok' do
+      subject
+
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'deletes the target' do
+      expect {
+        subject
+      }.to change { Target.count }.by(-1)
```
Comment (juanmferreira93) [inline spec/requests/api/v1/targets/destroy_spec.rb]:
```suggestion
      expect {
        subject
      }.to change { Target.count }.from(1).to(0)
```
It's more precise

**Review by juanmferreira93 — APPROVED** [review]
🚀

File: spec/requests/api/v1/targets/destroy_spec.rb [hunk 12]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+describe 'DELETE api/v1/targets/:id' do
+  let(:user) { create(:user) }
+  let(:headers) { auth_headers }
```
Comment (AgostinaDev) [inline spec/requests/api/v1/targets/destroy_spec.rb:5]: I think this is not necessary unless you're overriding the headers in one of the test cases (in a test for non-authenticated calls, for example)

File: spec/requests/api/v1/targets/destroy_spec.rb [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  context 'when user sends someone elses target id' do
+    let(:target_id) { -1 }
+
+    it 'returns not_found' do
+      subject
+      expect(response).to have_http_status(:not_found)
+    end
+  end
```
Comment (AgostinaDev) [inline spec/requests/api/v1/targets/destroy_spec.rb:32]: WDYT about adding tests for non-authenticated calls? 

Like this one you did for another endpoint:

https://github.com/LeandroSoria2/TargetApi/blob/34616fb9028971e126d037aadce4b8e083286207/spec/requests/api/v1/targets/index_spec.rb#L35

**Review by AgostinaDev — APPROVED** [review]
GJ 💪

**Pull request merged**